library-model/library-model-generator/src/main/java/com/uber/nullaway/libmodel/LibraryModelGenerator.java

```diff
@@ -288,13 +329,15 @@ private boolean hasNullableReturn(MethodDeclaration md) {
      * @param md The MethodDeclaration instance.
      * @return The return type string value to be written into the astubx file.
      */
-    private String getMethodReturnTypeString(MethodDeclaration md) {
-      if (md.getType() instanceof ClassOrInterfaceType) {
-        return md.getType().getChildNodes().get(0).toString();
-      } else if (md.getType() instanceof ArrayType) {
+    private String getMethodReturnTypeString(ResolvedMethodDeclaration md) {
+      ResolvedType returnType = md.getReturnType();
+      if (returnType.isReferenceType()) {
+        return returnType.asReferenceType().getQualifiedName().toString();
+      } else if (returnType.isArray()) {
         return ARRAY_RETURN_TYPE_STRING;
       } else {
-        return md.getType().toString();
+        throw new RuntimeException("Unexpected return type: " + returnType);
```

Comment:
We need a unit test where a method returns a primitive type (like `int`) and we need to handle it correctly (same as JarInfer)